logstash-core/api/lib/app/system/info_command.rb

```diff
@@ -4,20 +4,30 @@
 class LogStash::Api::SystemInfoCommand < LogStash::Api::Command
 
   def run
-    report = { "version"   => "1.0.0",
-               "host_name" => "foobar",
-               "plugins"   => { "count" => 10, "list" => [] }
+    plugins_list = available_plugins
+    {
+      "version"   => LOGSTASH_CORE_VERSION,
+      "hostname" => hostname,
+      "plugins"   => { "count" => plugins_list.count, "list" => plugins_list.sort }
     }
-    report
   end
 
   private
 
-  def type
-    ["input", "output", "filter"][rand(3)]
+  def hostname
+    `hostname`.strip
   end
```

Comment:
*I believe* `Socket.gethostbyname` doesn't do dns check, if you want to do that you have to use 
``
Socket.getbyhostname(Socket.gethostname)